src/main/java/org/openrewrite/java/spring/doc/MigrateDocketBeanToGroupedOpenApiBean.java

```diff
@@ -0,0 +1,304 @@
+        return Preconditions.check(or(isDocketJavaBeanConfiguration(), new IsPossibleSpringConfigFile()), new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public Tree visit(Tree tree, ExecutionContext ctx) {
+                if (tree instanceof J.CompilationUnit) {
+                    return new JavaBeanConfigurationScanner(acc).visit(tree, ctx);
+                } else if (tree != null && new IsPossibleSpringConfigFile().visit(tree, ctx).getMarkers().findFirst(SearchResult.class).isPresent()) {
+                    acc.hasProperties = true;
+                }
+                return super.visit(tree, ctx);
+            }
+        });
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor(DocketBeanAccumulator acc) {
+        if (acc.docketDefinitions.size() != 1) {
+            return TreeVisitor.noop();
+        }
+        DocketDefinition docketDefinition = acc.docketDefinitions.get(0);
+        return Preconditions.check(or(isDocketJavaBeanConfiguration(), new IsPossibleSpringConfigFile()), new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public @Nullable Tree visit(@Nullable Tree tree, ExecutionContext ctx, Cursor parent) {
+                if (tree instanceof J.CompilationUnit) {
+                    return new JavaIsoVisitor<ExecutionContext>() {
+                        @Override
+                        public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+                            J.MethodDeclaration md = super.visitMethodDeclaration(method, ctx);
+                            if (md.getLeadingAnnotations().stream().anyMatch(BEAN_ANNOTATIONMATCHER::matches) && DOCKET_TYPEMATCHER.matches(md.getReturnTypeExpression())) {
+                                maybeRemoveImport("org.springframework.context.annotation.Bean");
+                                maybeRemoveImport("springfox.documentation.builders.PathSelectors");
+                                maybeRemoveImport("springfox.documentation.builders.RequestHandlerSelectors");
+                                maybeRemoveImport("springfox.documentation.spi.DocumentationType");
+                                maybeRemoveImport("springfox.documentation.spring.web.plugins.Docket");
+                                if (canConfigureInProperties(acc, docketDefinition)) {
+                                    return null;
+                                } else {
+                                    String methodTemplate = "@Bean\n" +
+                                            "public GroupedOpenApi " + method.getSimpleName() + "() {\n" +
+                                            "    return GroupedOpenApi.builder()\n";
+                                    if (docketDefinition.groupName == null) {
+                                        methodTemplate = methodTemplate + "            .group(\"public\")\n";
```

Comment:
```suggestion
                                        methodTemplate += "            .group(\"public\")\n";
```

Also use the `+=` operator for the lines beneath.